core/src/main/java/com/scalar/db/transaction/singlecrudoperation/SingleCrudOperationTransactionManager.java

```diff
@@ -0,0 +1,364 @@
+  @Override
+  public void insert(Insert insert) throws CrudException {
+    insert = copyAndSetTargetToIfNot(insert);
+
+    PutBuilder.Buildable buildable =
+        Put.newBuilder()
+            .namespace(insert.forNamespace().orElse(null))
+            .table(insert.forTable().orElse(null))
+            .partitionKey(insert.getPartitionKey());
+    insert.getClusteringKey().ifPresent(buildable::clusteringKey);
+    insert.getColumns().values().forEach(buildable::value);
+    buildable.condition(ConditionBuilder.putIfNotExists());
+    Put put = buildable.consistency(Consistency.LINEARIZABLE).build();
+
+    try {
+      storage.put(put);
+    } catch (NoMutationException e) {
+      throw new CrudConflictException(
+          CoreError.SINGLE_CRUD_OPERATION_TRANSACTION_CONFLICT_OCCURRED_IN_INSERT.buildMessage(),
+          e,
+          null);
+    } catch (ExecutionException e) {
+      throw new CrudException(e.getMessage(), e, null);
+    }
+  }
+
+  @Override
+  public void upsert(Upsert upsert) throws CrudException {
+    upsert = copyAndSetTargetToIfNot(upsert);
+
+    PutBuilder.Buildable buildable =
+        Put.newBuilder()
+            .namespace(upsert.forNamespace().orElse(null))
+            .table(upsert.forTable().orElse(null))
+            .partitionKey(upsert.getPartitionKey());
+    upsert.getClusteringKey().ifPresent(buildable::clusteringKey);
+    upsert.getColumns().values().forEach(buildable::value);
+    Put put = buildable.consistency(Consistency.LINEARIZABLE).build();
+
+    try {
+      storage.put(put);
+    } catch (ExecutionException e) {
+      throw new CrudException(e.getMessage(), e, null);
+    }
+  }
+
+  @Override
+  public void update(Update update) throws CrudException {
+    update = copyAndSetTargetToIfNot(update);
+
+    ScalarDbUtils.checkUpdate(update);
+    PutBuilder.Buildable buildable =
+        Put.newBuilder()
+            .namespace(update.forNamespace().orElse(null))
+            .table(update.forTable().orElse(null))
+            .partitionKey(update.getPartitionKey());
+    update.getClusteringKey().ifPresent(buildable::clusteringKey);
+    update.getColumns().values().forEach(buildable::value);
+    if (update.getCondition().isPresent()) {
+      if (update.getCondition().get() instanceof UpdateIf) {
+        buildable.condition(ConditionBuilder.putIf(update.getCondition().get().getExpressions()));
+      } else {
+        assert update.getCondition().get() instanceof UpdateIfExists;
+        buildable.condition(ConditionBuilder.putIfExists());
+      }
+    } else {
+      buildable.condition(ConditionBuilder.putIfExists());
+    }
+    Put put = buildable.consistency(Consistency.LINEARIZABLE).build();
+
+    try {
+      storage.put(put);
+    } catch (NoMutationException e) {
+      if (update.getCondition().isPresent()) {
+        throwUnsatisfiedConditionException(update);
+      }
+
+      // If the condition is not specified, it means that the record does not exist. In this case,
```

Comment:
Sorry, I am confused by this comment. 
If the `update` condition is not specified, a `PutIfExists` condition is set. If the record already exists, the `NoMutationException` will be thrown, and we can ignore it, isn't it?

Reply:
Sorry for the confusion.

If the `update` condition is not specified, we specify `ConditionBuilder.putIfExists()` for the put operation here:
https://github.com/scalar-labs/scalardb/pull/1793/files/aecdf951b623ea4d77be27435ca397eddb5414ce#diff-ca66c3466e760469ea1c0982cf3eb91f275cf80e0c5a7e808085afb8296374e5R255

Therefore, if a `NoMutationException` is thrown and the `update` condition is not specified, it means that the record doesn't exist. Does this explanation make sense?